ocs_ci/ocs/replica_one.py

```diff
@@ -0,0 +1,199 @@
+        ):
+            if deployment["metadata"]["labels"]["app.kubernetes.io/name"] == "ceph-osd":
+                osd_deployment.append(deployment)
+
+    for deployment in osd_deployment:
+        if (
+            deployment["metadata"]["labels"]["ceph.rook.io/DeviceSet"]
+            in config.ENV_DATA["worker_availability_zones"]
+        ):
+            log.info(deployment["metadata"]["name"])
+            replica1_osd_deployment.append(deployment["metadata"]["name"])
+
+    return replica1_osd_deployment
+
+
+def scaledown_deployment(deployments_name=list[str]) -> None:
+    """
+    Scale down deployments to 0
+
+    Args:
+        deployments_name (list[str]): list of deployment names.
+
+    """
+    deployment_obj = OCP(kind=DEPLOYMENT)
+    for deployment in deployments_name:
+        deployment_obj.exec_oc_cmd(f"scale deployment {deployment} --replicas=0")
+        log.info(f"scaling to 0: {deployment}")
+
+
+def count_osd_pods() -> int:
+    """Get OSDs count in a cluster
+
+    Returns:
+        int: number of OSDs in cluster
+
+    """
+    return len(get_pods_having_label(label=OSD_APP_LABEL))
+
+
+def delete_replica_1_sc() -> None:
+    """
+    Deletes storage class associated with replica1
+
+    """
+    sc_obj = OCP(kind=STORAGECLASS, resource_name=REPLICA1_STORAGECLASS)
+    try:
+        sc_obj.delete(resource_name=REPLICA1_STORAGECLASS)
+    except CommandFailed as e:
+        if "Error is Error from server (NotFound)" in str(e):
+            log.info(
+                f"{REPLICA1_STORAGECLASS} not found, assuming it was already deleted"
+            )
+        else:
+            log.error("Failed to delete storage class")
+
+
+def purge_replica1_osd():
+    """
+    Purge OSDs associated with replica1
+        1. scale down its deployments to 0
+        2. use OSD removal template
+
+    """
+    deployments_name = get_replica1_osd_deployment()
+    scaledown_deployment(deployments_name)
+    (osds, osds_id) = get_replica_1_osds()
+    log.info(f"OSD IDs: {osds}")
+    log.info(f"OSD IDs: {osds_id}")
+    run_osd_removal_job(osd_ids=osds_id)
+
+
+def delete_replica1_cephblockpools_cr(cbp_object: OCP):
+    """
+    Deletes CR of cephblockpools associated with replica1
+
+    Args:
+        cbp_object (OCP): OCP object with kind=CEPHBLOCKPOOL
+
+    """
+    for i in range(0, len((cbp_object.data["items"]))):
+        cbp_cr_name = cbp_object.data["items"][i]["metadata"]["name"]
+        if cbp_cr_name != DEFAULT_CEPHBLOCKPOOL:
+            cbp_object.delete(resource_name=cbp_cr_name)
+
+
+def delete_replica1_cephblockpools(cbp_object: OCP):
+    """
+    Deletes cephblockpools associated with replica1
+
+    Args:
+        cbp_object (OCP): OCP object with kind=CEPHBLOCKPOOL
+
+    """
+    toolbox_pod = get_ceph_tools_pod()
+    for i in range(0, len((cbp_object.data["items"]))):
+        replica1_pool_name = cbp_object.data["items"][i]["metadata"]["name"]
+        if replica1_pool_name != DEFAULT_CEPHBLOCKPOOL:
```

Comment:
same here

Reply:
removed this function as it is not needed anymore